modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -696,11 +697,41 @@ export default {
           body: {},
           busy: true
         });
-        apos.notify(response ? 'Restored previously published version.' : 'No longer published.', {
+        apos.notify('Restored previously published version.', {
           type: 'success',
           dismiss: true
         });
-        // This handler covers all "undo publish" buttons, so make sure it's
+        // This handler covers the modals too, so make sure it's
+        // for the context document before altering any admin bar state
+        // because of it
+        if (data._id.replace(/:.*$/, '') === (this.context._id.replace(/:.*$/, ''))) {
+          this.context = {
+            ...this.context,
+            modified: true,
+            lastPublishedAt: response && response.lastPublishedAt
+          };
+          // No refresh is needed here because we're still in draft mode
+          // looking at the draft mode, and the thing that changed is the
+          // published mode
+        }
+      } catch (e) {
+        await apos.alert({
+          heading: 'An Error Occurred',
+          description: e.message || 'An error occurred while restoring the previously published version.'
+        });
+      }
+    },
+    async onUnpublish(data) {
+      try {
```

Comment:
The catch message on this is 'An error occurred while restoring the previously published version.' That should be updated for the `onUnpublish` method.